pkg/objstore/azure/helpers.go

```diff
@@ -64,6 +116,28 @@ func getContainerURL(ctx context.Context, conf Config) (blob.ContainerURL, error
 	return service.NewContainerURL(conf.ContainerName), nil
 }
 
+func DefaultTransport(config Config) *http.Transport {
```

Comment:
That sounds like something we can reuse across buckets.

Reply:
I would like to propose to not do this in this PR but create a new issue for this. As this would mean a small refactor in various parts in Thanos. This makes the scope of this PR too big IMO. WDYT?